src/v/cluster/partition.cc

```diff
@@ -843,23 +843,15 @@ void partition::set_topic_config(
 }
 
 ss::future<std::error_code>
-partition::transfer_leadership(std::optional<model::node_id> target) {
+partition::transfer_leadership(transfer_leadership_request req) {
```

Comment:
Why is a `req` passed in here now? Is `req.group` ignored?

Reply:
It'd be ignored. I was just passing the whole request so that `timeout` or any future fields could be pulled out of it if needed. The other option is to pass the target and `raft::transfer_leadership_options` if that seems like a cleaner design.

Reply:
SGTM.